enhancements/cluster-logging/performance-tuning.md

```diff
@@ -0,0 +1,238 @@
+Achieving a performance goal may involve configuring multiple parts of the delivery pipeline consistently.
+This would be difficult and error prone for users, but it can be safely automated by our configuration generator.
+
+### User Stories
+#### As a cluster logging administrator I want to minimize log loss.
+
+``` yaml
+outputs:
+ new - name: minimise-log-loss
+   tuning:
+     delivery: safe
+```
+
+#### As a cluster logging administrator I want to maximize throughput.
+
+``` yaml
+outputs:
+  - name: maximise-throughput
+    tuning:
+      delivery: fast
+```
+
+#### As a cluster logging administrator I want to tweak features of a specific output.
+
+``` yaml
+outputs:
+  - name: detailed-tweaks
+    tuning:
+	  delivery: safe
+	  compress: true  # Enable protocol compression
+	  reconnect:
+	    maxDelay: 1s  # Wait max of 1 second between reconnect attempts
+	  batch:
+		maxBytes: 50K # Send max 50K of data in each write.
+```
+
+### GOALS
+
+- Accomodate users with different performance and reliability trade-offs.
+- Allow users to tune output parameters that we cannot optimize automatically.
+- Expose a simple, general purpose, collector-neutral tuning API
+- Use vector's end-to-end acknowledgements for more efficient log delivery via vector.
+
+### Non-Goals
+
+- Not offering guaruanteed reliable end-to-end delivery (limited by reliability of source and sink)
+- Not exposing underlying vector configuration details.
+- No rate limiting - already provided by `outputs[].limit` field.
+- No plans to support fluentd.
+
+## Proposal
+
+### Workflow Description
+
+**logging administrator** is a human user responsible for setting output tuning parameters as needed.
+
+### API Extensions
+
+New field `ClusterLogForwarder.output[].tuning` with the following sub-fields:
+
+#### delivery: Fast|Safe
+
+`Fast`: Minimize resource use (CPU, disk, network) and maximize throughput in normal operation. Use when:
+- It is acceptable to lose some logs due to restarts, failures or heavy load.
+- The logging system will run close to it's limits, and may sometimes be overloaded.
+- It is important to minimize memory, CPU, disk and network costs.
+
+`Safe`: Minimize log loss due to restarts or failures. Use when:
+- It is important to avoid log loss due to restarts or failures.
+- The logging system is not expected to be pushed close to its limits.
+- Resources (memory, CPU, disk) are sufficient for log throughput to exceed expected peak load.
+
+**Note**: The log collector _cannot_ guarantee reliable end-to-end delivery.
+It has no control over the reliability or throughput of sources and sinks.
+The `delivery` policy tunes the log collector's "best effort".
+
+#### compress: string
+
+Enable compression if supported
+- Value is compression "mode" string, format depends on output type.
+- All outputs accept `compress: true` to enable a default mode.
+- Error if output does not support compression or does not recognize the value.
+
+#### batch: object
```

Comment:
Logically separate concern. Persistent buffer is for recovery from disconnects. Batch size is max data per individual "write". Will review vector buffering to see if they clash in the implementation...

Reply:
My point to mention it here is that we have the fast/slow setting but we then also offer a way to override the  those.  There is no place to override and set persistent buffering

Reply:
fast implies in-memory. safe implies persistent buffers or end-to-end ack, whichever is most efficient given the source/sink combinations. Do we need more overrides?

Reply:
Please allow me to join this discussion abruptly, and let me share my thoughts.

One thing that comes to my mind as a concern is that, as indicated in "https://github.com/vectordotdev/vector/discussions/12991", 
end-to-end (e2e) support is not available in Vector for the combination of using Lua transfer (source -> lua transfer -> sink). Lua Transfer is used a part of OpenShift Logging.
Therefore it might be difficult to automatically determine e2e or Persistent buffer from only the combination of source and sink.

Reply:
@kattz-kawa we have a desire to refactor the lua transform and replace it with VRL.  I don't recall if we have a ticket to do that but I believe everything we need to do can now be accomplished with VRL.  If that is only blocker in your comment then we should proceed to prioritize the refactoring

Reply:
@jcantrill 
thank you for the reply. 
Remap transfer is supported for e2e, as far as I can see. 
so refactoring the lua transform is fine for my concern.

Reply:
@alanconway in summary of our f2f discussion, initial pass forgoes implicit disk buffering to use e2e acks which makes the original source "the buffer"